staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/unstructured.go

```diff
@@ -282,9 +319,14 @@ func (u *Unstructured) GetCreationTimestamp() metav1.Time {
 }
 
 func (u *Unstructured) SetCreationTimestamp(timestamp metav1.Time) {
+	// creationTimestamp is a struct (metav1.Time) so omitempty holds no meaning for it.
```

Comment:
if this isn't causing issues with etcd resourceVersion changes (which it shouldn't, since we don't persist objects with zero-value creationTimestamps to etcd), let's not make this change in this PR. I'd actually like to see if it would be possible to go the other direction in a separate PR, and make ObjectMeta actually omit creationTimestamp when empty.

Reply:
> if this isn't causing issues with etcd resourceVersion changes 

it doesn't :+1: 

> make ObjectMeta actually omit creationTimestamp when empty.

My understanding is that `omitEmpty` does not apply to structs and no matter what the value of `creationTimestamp` is, this should persist in the object.

Reply:
> My understanding is that omitEmpty does not apply to structs and no matter what the value of creationTimestamp is, this should persist in the object.

You are correct. I'd like to explore switching the creationTimestamp to a `*time.Time` so it can actually be omitted (verifying wire compatibility for existing json and proto, obviously)

Reply:
> I'd like to explore switching the creationTimestamp to a *time.Time so it can actually be omitted (verifying wire compatibility for existing json and proto, obviously)

created https://github.com/kubernetes/kubernetes/issues/67610 to track this.